Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs

```diff
@@ -147,13 +150,44 @@ internal static void CheckForDefaultUIActionMapChanges()
             }
         }
 
-        #endif
+#endif
+        /// <summary>
+        /// Reset project wide input actions asset
+        /// </summary>
+        internal static void ResetActionAsset()
+        {
+            CreateNewActionAsset();
+        }
+
+        /// <summary>
+        /// Delete project wide input actions
+        /// </summary>
+        internal static void DeleteActionAssetAndActionReferences()
+        {
+            var objects = AssetDatabase.LoadAllAssetsAtPath(s_AssetPath);
+            if (objects != null)
+            {
+                // Handle deleting all InputActionAssets as older 1.8.0 pre release could create more than one project wide input asset in the file
+                foreach (var obj in objects)
+                {
+                    if (obj is InputActionReference)
+                    {
+                        var actionReference = obj as InputActionReference;
+                        actionReference.Set(null);
+                        AssetDatabase.RemoveObjectFromAsset(obj);
+                    }
+                    else if (obj is InputActionAsset)
+                    {
+                        AssetDatabase.RemoveObjectFromAsset(obj);
+                    }
```

Comment:
What happens if there is something here that isn't an InputActionReference or an InpuitAction asset?
I recommend we either throw or log an error or alternatively change the last else if into an unconditional else which would remove anything? Or is this designed to leave e.g. InputManager or any other root objects untouched?

Reply:
Its designed to leave InputManager and others untouched